ddl/error.go

```diff
@@ -191,6 +191,14 @@ var (
 	ErrFunctionalIndexOnField = terror.ClassDDL.New(mysql.ErrFunctionalIndexOnField, mysql.MySQLErrName[mysql.ErrFunctionalIndexOnField])
 	// ErrInvalidAutoRandom returns when auto_random is used incorrectly.
 	ErrInvalidAutoRandom = terror.ClassDDL.New(mysql.ErrInvalidAutoRandom, mysql.MySQLErrName[mysql.ErrInvalidAutoRandom])
+	// ErrUnsupportedColumnCheck returns CHECK for column is not supported when create table
+	ErrUnsupportedColumnCheck = terror.ClassDDL.New(mysql.ErrUnsupportedConstraintCheck, fmt.Sprintf(mysql.MySQLErrName[mysql.ErrUnsupportedConstraintCheck], "Column check"))
+	// ErrUnsupportedAlterCheck returns when use ALTER TABLE ALTER CHECK is not supported
+	ErrUnsupportedAlterCheck = terror.ClassDDL.New(mysql.ErrUnsupportedConstraintCheck, fmt.Sprintf(mysql.MySQLErrName[mysql.ErrUnsupportedConstraintCheck], "ALTER CHECK"))
+	// ErrUnsupportedDropCheck returns when ALTER TABLE DROP CHECK is not supported
+	ErrUnsupportedDropCheck = terror.ClassDDL.New(mysql.ErrUnsupportedConstraintCheck, fmt.Sprintf(mysql.MySQLErrName[mysql.ErrUnsupportedConstraintCheck], "DROP CHECK"))
+	// ErrUnsupportedConstraintCheck returns when use ADD CONSTRAINT CHECK
+	ErrUnsupportedConstraintCheck = terror.ClassDDL.New(mysql.ErrUnsupportedConstraintCheck, fmt.Sprintf(mysql.MySQLErrName[mysql.ErrUnsupportedConstraintCheck], "ADD CONSTRAINT CHECK"))
```

Comment:
I mean we could only have exactly one error. 
`ErrUnsupportedColumnCheck`, `ErrUnsupportedAlterCheck` and  `ErrUnsupportedDropCheck` can be represented by `ErrUnsupportedConstraintCheck`.

Reply:
I've noticed that the interface for adding warning is `func (sc *StatementContext) AppendWarning(warn error)` which accept error as an argument. If there is only one `ErrUnsupportedConstraintCheck` I cannot figure out a way to custom the "%s" is not supported. 
Another solutions might be dynamically new an error type or using only one error info like "CHECK is not supported". But I think this is not very clear, so eventually I wrote 4 error types to specify 4 cases.
Is there any way I could append warning by providing extra args with exactly one error type?

Reply:
How about `GenWithStackByArgs`:
```go
ctx.GetSessionVars().StmtCtx.AppendWarning(ErrUnsupportedConstraintCheck.GenWithStackByArgs("Column check constraint"))
```

Reply:
> How about `GenWithStackByArgs`:
> 
> ```go
> ctx.GetSessionVars().StmtCtx.AppendWarning(ErrUnsupportedConstraintCheck.GenWithStackByArgs("Column check constraint"))
> ```

This is helpful, I've modified the code, now we only have *1* error type `ErrUnsupportedConstraintCheck`